scout/server/blueprints/diagnoses/controllers.py

```diff
@@ -25,7 +25,22 @@ def disease_entry(store, disease_id) -> dict:
 
 
 def disease_terms(store: MongoAdapter, query: str, source: str) -> dict:
-    """Retrieve disease terms, queried by source or a text-match for the disease description."""
+    """Retrieve disease terms, queried by source or a text-match for the disease description.
+    disease_term[genes] are populated with both hgnc_id and symbol"""
+
+    disease_data = list(store.query_disease(query=query, source=source, filter_project=None))
+
+    for disease in disease_data:
+        gene_ids = disease.get("genes", [])
+        gene_symbols = []
+
+        for gene in gene_ids:
+            gene_symbol = store.hgnc_gene_caption(hgnc_identifier=gene)
```

Comment:
```suggestion
            gene_caption: dict = store.hgnc_gene_caption(hgnc_identifier=gene)
```

Reply:
It's not a gene symbol, but a dictionary. If I read symbol immagine it as a string or a dictionary like {hgnc_symbol:"LAMA1"}

Reply:
Aaah, you are right, I believe it was in fact a string when I named it before realising I would also need the id and replaced the function 😆 The variable names have been replaced as suggested